http/sys_raft.go

```diff
@@ -0,0 +1,50 @@
+package http
+
+import (
+	"context"
+	"io"
+	"net/http"
+
+	"github.com/hashicorp/vault/vault"
+)
+
+func handleSysRaftJoin(core *vault.Core) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.Method {
+		case "POST", "PUT":
+			handleSysRaftJoinPost(core, w, r)
+		default:
+			respondError(w, http.StatusMethodNotAllowed, nil)
+		}
+	})
+}
+
+func handleSysRaftJoinPost(core *vault.Core, w http.ResponseWriter, r *http.Request) {
+	// Parse the request
+	var req JoinRequest
+	if _, err := parseRequest(core, r, w, &req); err != nil && err != io.EOF {
+		respondError(w, http.StatusBadRequest, err)
+		return
+	}
+
+	joined, err := core.JoinRaftCluster(context.TODO(), req.LeaderAddr, nil, req.Retry)
```

Comment:
I assume this should be context.Background()? Since we won't have c.activeContext yet.